dmoj/graders/bridged.py

```diff
@@ -0,0 +1,151 @@
+                # So we set it to None because we don't need to access it
+                result = Result(None)
+                result.set_result_flag(interactor)
+                if hasattr(interactor_binary, 'feedback') and interactor.feedback:
+                    feedback = interactor.feedback
+                else:
+                    feedback = getattr(interactor_binary, 'get_feedback', lambda x, y, z: '')(stderr, result, interactor)
+                if feedback:
+                    error += ' with feedback: %s' % feedback
+            raise InternalError(error)
+
+    def _interact_with_process(self, case, result, input, output=None):
+        with mktemp(input) as input_file, mktemp(output) as output_file:
+            submission_stdin, interactor_stdout = os.pipe()
+            interactor_stdin, submission_stdout = os.pipe()
+
+            self.interactor = self.interactor_binary.launch(input_file.name, output_file.name,
+                                                            time=(self.handler_data.preprocessing_time or 0) +
+                                                            self.problem.time_limit,
+                                                            memory=self.handler_data.memory_limit or
+                                                            env['generator_memory_limit'],
+                                                            stdin=interactor_stdin,
+                                                            stdout=interactor_stdout,
+                                                            stderr=subprocess.PIPE)
+
+            self._current_proc = self.binary.launch(time=self.problem.time_limit,
+                                                    memory=self.problem.memory_limit,
+                                                    symlinks=case.config.symlinks,
+                                                    stdin=submission_stdin,
+                                                    stdout=submission_stdout,
+                                                    stderr=subprocess.PIPE,
+                                                    wall_time=case.config.wall_time_factor * self.problem.time_limit)
+
+            os.close(submission_stdout)
+            os.close(submission_stdin)
+            os.close(interactor_stdout)
+            os.close(interactor_stdin)
+
+            self._current_proc.wait()
+            self.interactor.wait()
+
+            return self._current_proc.stderr.read()
+
+    def _generate_interactor_binary(self):
+        files = self.handler_data.files
+        if not isinstance(files, list):
+            files = [files]
+        files = [os.path.join(get_problem_root(self.problem.id), f) for f in files]
+        return compile_with_auxiliary_files(files, self.handler_data.lang,
+                                            self.handler_data.compiler_time_limit or env['generator_compiler_limit'])
```

Comment:
don't reuse random time limits for something unrelated; isn't `compiler_time_limit` defaulted anyway?